### PR TITLE
Update next link implementation

### PR DESCRIPTION
**Pull request by mucktar13**
# Update next link implementation
- [ ] JIRA Issue ID (preferably only ONE parent issue, put subtask IDs in the Description)
- [ ] Multiple JIRA Issue IDs? Are you sure your PR isn't too big? Can you rebase your commits to new branches so there's only 1 issue ID?
- [ ] Is your title relatable to the JIRA issue?
- [ ] Title <= 100 characters

# Description

reference 
https://github.com/vercel/next.js/discussions/12578#discussioncomment-11375
https://www.loudnoises.us/solving-next-static-pages-js-404-not-found/

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] Builds and runs on Chrome Desktop
- [ ] Builds and runs on Chrome Responsive
- [ ] Unit tests?
- [ ] Integration tests?
- [ ] End-to-end (E2E) tests?

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Have you added yourself as the Assignee?
- [ ] Have you added 1 or more leads as the Reviewer?
- [ ] Have you chosen 1-2 of the following labels?
  - Feature
  - Bug
  - Refactor
  - Test
  - Documentation

### CODE REVIEW

File: src/components/PostEditor/PostRender.tsx [hunk 1]
```diff
@@ -129,78 +133,90 @@ export const PostRender: React.FC<PostRenderProps> = props => {
...
-                  style={{
-                    cursor: 'pointer',
-                    fontWeight: 600,
-                    color: theme.palette.primary.main,
-                    display: 'inline-block',
-                  }}>
-                  #{node.hashtag}
-                </Typography>
-              </a>
+            <Link key={id} href={`/topic/hashtag?tag=${node.hashtag}`} shallow passHref>
```
Comment (ical10) [inline src/components/PostEditor/PostRender.tsx:184]: add `as={....}` syntax

File: src/components/Trending/TrendingList.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const TrendingList: React.FC<TrendingListProps> = props => {
         {trendings.map((trending, i) => (
           <ListItem className={style.item} key={uuid()}>
             <ListItemText disableTypography className={style.text}>
-              <Link href={`/topic/hashtag?tag=${trending.id}`} shallow={false}>
+              <Link href={`/topic/hashtag?tag=${trending.id}`} shallow>
```
Comment (ical10) [inline src/components/Trending/TrendingList.tsx:33]: add `as={....}` params

**Review by ical10 — CHANGES_REQUESTED** [review]

File: src/components/Trending/TrendingList.tsx (hunk 2 above)
Comment (mucktar13) [reply src/components/Trending/TrendingList.tsx:33]: ini ga dynamic bro @ical10

File: src/components/PostEditor/PostRender.tsx (hunk 1 above)
Comment (mucktar13) [reply src/components/PostEditor/PostRender.tsx:184]: @ical10 ini ga dinamyc juga karna query param

**Review by ical10 — APPROVED** [review]

**Pull request merged**